application/client/src/app/ui/elements/transport/serial/component.ts

```diff
@@ -1,21 +1,67 @@
-import { Component, ChangeDetectorRef, Input } from '@angular/core';
-import { Ilc, IlcInterface } from '@env/decorators/component';
+import { Component, ChangeDetectorRef, Input, OnDestroy, AfterViewInit } from '@angular/core';
 import { ChangesDetector } from '@ui/env/extentions/changes';
 import { State } from './state';
 import { Action } from '@ui/tabs/sources/common/stream.open/action';
+import { Ilc, IlcInterface } from '@env/decorators/component';
+
+import SerialService from './service/service';
 
 @Component({
     selector: 'app-transport-serial',
     templateUrl: './template.html',
     styleUrls: ['./styles.less'],
 })
 @Ilc()
-export class TransportSerial extends ChangesDetector {
+export class TransportSerial extends ChangesDetector implements AfterViewInit, OnDestroy {
     @Input() public state!: State;
     @Input() public action!: Action;
 
+    public _ng_ports: string[] = [];
+
+    private _timeout!: number;
+
     constructor(cdRef: ChangeDetectorRef) {
         super(cdRef);
     }
+
+    public ngAfterViewInit() {
+        this._detectPorts();
+    }
+
+    public ngOnDestroy() {
+        clearTimeout(this._timeout);
+        this.state.destroy();
+    }
+
+    private _detectPorts() {
+        this.ilc()
+            .services.system.bridge.ports()
+            .ports()
+            .then((ports: string[]) => {
+                this._ng_ports = ports;
+                let selected: string = SerialService.selected;
+                if (this._ng_ports.indexOf(SerialService.selected) === -1) {
+                    selected =
+                        this._ng_ports.length > 0
+                            ? SerialService.setSelected(this._ng_ports[0])
+                            : SerialService.setSelected('');
+                }
+                if (this._ng_ports.length === 0) {
+                    selected = SerialService.setSelected('');
+                } else if (SerialService.selected === '' && this._ng_ports.length > 0) {
+                    selected = SerialService.setSelected(this._ng_ports[0]);
+                }
+                this.state.path = selected;
+                this.action.setDisabled(selected === '');
+            })
+            .catch((err: Error) => {
+                console.log(`Fail to update ports list due error: ${err.message}`);
```

Comment:
`console.log(...)` only for debugging )
use please `this.log().error(...)`, it comes with `ilc`